enterprise/ha/src/main/java/org/neo4j/kernel/ha/id/HaIdGeneratorFactory.java

```diff
@@ -73,6 +73,7 @@ public IdGenerator open( FileSystemAbstraction fs, File fileName, int grabSize,
             initialIdGenerator = localFactory.open( fs, fileName, grabSize, idType, highId );
             break;
         case SLAVE:
+            fs.deleteFile( fileName );
```

Comment:
The first time we seem to open _after_ a call to `switchToSlave`, so this is needed, although it probably could be done by observing current state and invoke correct `switchToXXX` method in open.